fl4health/clients/basic_client.py

```diff
@@ -1105,6 +1144,37 @@ def get_model(self, config: Config) -> nn.Module:
         """
         raise NotImplementedError
 
+    def get_lr_scheduler(self, optimizer_key: str, config: Config) -> Union[None, _LRScheduler]:
```

Comment:
I think perhaps, in a future iteration, we should use the get/set paradigm that we've used for the optimizer construction/initialization? That is, we have a `get_lr_scheduler` function that the user overrides to define the LRs that they want to use. The `set_lr_schedulers` function is called by the `setup_client` function and invokes the `get_lr_scheduler` method. The reason I like this is because the `set_lr_schedulers` function can do sanity checks by default and encapsulate the behavior you have above. That is,
```
def set_lr_schedulers(self, config: Config) -> Optional[_LRScheduler]:
       # Ensure that the optimizers have been initialized prior to setting schedulers
       assert self.optimizers
        # set schedulers for the optimizers
        self.lr_schedulers = {}
        for optimizer_key in self.optimizers.keys():
            lr_scheduler = self.get_lr_scheduler(optimizer_key, config)
            if lr_scheduler is not None:
                self.lr_schedulers[optimizer_key] = lr_scheduler
```